### PR TITLE
New decorator to add deadline prop to project application tasks

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

File: lib/decorators/deadline.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const { get } = require('lodash');
+const moment = require('moment-business-time');
+const Case = require('@ukhomeoffice/taskflow/lib/models/case');
```
Comment (wheelsandcogs) [inline lib/decorators/deadline.js]: Is there a better way to get access to the Case model? This seems to work without requiring any further connection setup, but should we be passing it in instead?

File: lib/decorators/deadline.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    const isAmendment = get(c, 'data.modelData.status') === 'active';
+
+    if (isAmendment) {
+      return c;
+    }
+
+    return Promise.resolve()
+      .then(() => Case.find(c.id))
```
Comment (wheelsandcogs) [inline lib/decorators/deadline.js]: we re-fetch the case to get the activity log, which won't be available on `c` if a list of tasks is being decorated.

File: lib/decorators/deadline.js (hunk 1 above)
Comment (lennym) [reply lib/decorators/deadline.js]: There _should_ be a better way. This way is bad. 

Let me have a look.

File: lib/decorators/deadline.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const { get } = require('lodash');
+const moment = require('moment-business-time');
+const Case = require('@ukhomeoffice/taskflow/lib/models/case');
+
+module.exports = settings => {
+  return c => {
+    if (get(c, 'data.model') !== 'project') {
```
Comment (lennym) [inline lib/decorators/deadline.js]: This should also check that the action is a `grant`

**Comment by lennym** [comment]
It probably makes sense to do the "complete and correct" check here as well, so that we don't need to do the extra data lookups and traversals in the cases where it's not complete and correct, and so the deadline isn't relevant.

File: lib/decorators/deadline.js (hunk 1 above)
Comment (lennym) [reply lib/decorators/deadline.js]: Looks like there's already 4 other instances of this in the project, so if I wanted to have feelings about it then it's too late.

Crack on!

**Comment by lennym** [comment]
This looks good, but I'm going to defer merging it because I want to get the deployment up today and could do with having as few other moving parts as possible.

**Comment by wheelsandcogs** [comment]
👍

**Comment by lennym** [comment]
Can we rip a load of code out of the task screens now?

**Pull request merged**